### PR TITLE
Handle dependencies when reusing workloads on initial startup

### PR DESCRIPTION
**Pull request by lingnoi**
Issues: #261

# Definition of Done

The PR shall be merged only if all items mentioned in [CONTRIBUTING.md](https://github.com/eclipse-ankaios/ankaios/blob/main/CONTRIBUTING.md#how-to-contribute) have been followed. In case an item is not applicable as described, please provide a short explanation in the description.

- [x] All steps in [CONTRIBUTING.md](https://github.com/eclipse-ankaios/ankaios/blob/main/CONTRIBUTING.md#how-to-contribute) have been handled

### CODE REVIEW

File: agent/doc/swdesign/README.md [hunk 1]
```diff
@@ -671,6 +672,23 @@ Needs:
...
+##### RuntimeFacade deletes old workload without sending workload states
+`swdd~agent-delete-old-workload-without-sending-workload-states~1`
+
+Status: approved
+
+When the RuntimeFacade is requested to delete the workload and the flag `report_workload_states_for_workload` is false, then the RuntimeFacade shall delete the workload via the runtime connector without sending workload states.
+
+Comment:
+This delete is done by the specific runtime for a workload Id. No internal workload object is involved in this action.
+
```
Comment (krucod3) [inline agent/doc/swdesign/README.md:684]: Maybe we can add a rationale stating that this is a workaround needed until a dedicated workload restart handling is implemented.

File: agent/doc/swdesign/README.md [hunk 2]
```diff
@@ -658,7 +659,7 @@ Status: approved
 
 When requested, the RuntimeFacade shall delete a workload by:
```
Comment (maturar) [inline agent/doc/swdesign/README.md:660]: The sentence sound incomplete to me. What about write it in the same way as the new requirement:

```suggestion
When the RuntimeFacade is requested to delete the workload, then the RuntimeFacade shall delete a workload by:
```

File: agent/doc/swdesign/README.md (hunk 1 above)
Comment (inf17101) [reply agent/doc/swdesign/README.md:684]: I have added a rationale.

Comment (maturar) [reply agent/doc/swdesign/README.md:684]: Fixed

**Review by maturar — APPROVED** [review]
LGTM 👍

**Review by krucod3 — COMMENTED** [review]
I reviewed the swdesign/README.md. LGTM 👍

**Pull request merged**